pkg/gadgets/traceloop/syscall-helpers/syscall_helpers.go

```diff
@@ -219,27 +193,27 @@ func gatherSyscallsDeclarations() (map[string]syscallDeclaration, error) {
 	return cSyscalls, nil
 }
 
-func getSyscallDeclaration(syscallsDeclarations map[string]syscallDeclaration, syscallName string) (syscallDeclaration, error) {
+func GetSyscallDeclaration(syscallsDeclarations map[string]SyscallDeclaration, syscallName string) (SyscallDeclaration, error) {
 	declaration, ok := syscallsDeclarations[syscallName]
 	if !ok {
-		return syscallDeclaration{}, fmt.Errorf("no syscall correspond to %q", syscallName)
+		return SyscallDeclaration{}, fmt.Errorf("no syscall correspond to %q", syscallName)
 	}
 
 	return declaration, nil
 }
 
-func (s syscallDeclaration) getParameterCount() uint8 {
+func (s SyscallDeclaration) GetParameterCount() uint8 {
 	return uint8(len(s.params))
 }
 
-func (s syscallDeclaration) paramIsPointer(paramNumber uint8) (bool, error) {
+func (s SyscallDeclaration) ParamIsPointer(paramNumber uint8) (bool, error) {
 	if int(paramNumber) >= len(s.params) {
 		return false, fmt.Errorf("param number %d out of bounds for syscall %q", paramNumber, s.name)
 	}
 	return s.params[paramNumber].isPointer, nil
 }
 
-func (s syscallDeclaration) getParameterName(paramNumber uint8) (string, error) {
+func (s SyscallDeclaration) GetParameterName(paramNumber uint8) (string, error) {
```

Comment:
Related to my previous comment. I think we can make `param` and all its fields public, then this function won't be needed (specially important regarding the error handling around). The caller code could be something like:

```golang 
			syscallDeclaration, err := syscallhelpers.GetSyscallDeclaration(syscallsDeclarations, event.Syscall)
			if err != nil {
				return nil, fmt.Errorf("getting syscall definition")
			}

			for i, param := range syscallDeclaration.Params {
				fmt.Printf("param in position %d is named %s and is a pointer: %t\n", i, param.Name, param.IsPointer)
			}
```

WDYT?